### ISSUE TITLE
Cannot build germlines [sent to opig@stats.ox.ac.uk]

### ISSUE THREAD

**Issue by broncio123** [oxpig/ANARCI#31]
```
Hi,

 

I am trying to install ANARCI on ubuntu system with Python 2.7 and HMMER installed.

I ran the command: python2.7 setup.py install

Please see below the output, at the end it gives the error ( i have highligh:

running install

running build

running build_py

creating build

creating build/lib.linux-x86_64-2.7

creating build/lib.linux-x86_64-2.7/anarci

copying lib/python/anarci/schemes.py -> build/lib.linux-x86_64-2.7/anarci

copying lib/python/anarci/__init__.py -> build/lib.linux-x86_64-2.7/anarci

copying lib/python/anarci/anarci.py -> build/lib.linux-x86_64-2.7/anarci

creating build/lib.linux-x86_64-2.7/anarci/Bio

copying lib/python/anarci/Bio/SeqRecord.py -> build/lib.linux-x86_64-2.7/anarci/Bio

copying lib/python/anarci/Bio/__init__.py -> build/lib.linux-x86_64-2.7/anarci/Bio

copying lib/python/anarci/Bio/File.py -> build/lib.linux-x86_64-2.7/anarci/Bio

copying lib/python/anarci/Bio/bgzf.py -> build/lib.linux-x86_64-2.7/anarci/Bio

copying lib/python/anarci/Bio/Seq.py -> build/lib.linux-x86_64-2.7/anarci/Bio

copying lib/python/anarci/Bio/_utils.py -> build/lib.linux-x86_64-2.7/anarci/Bio

creating build/lib.linux-x86_64-2.7/anarci/Bio/Align

copying lib/python/anarci/Bio/Align/AlignInfo.py -> build/lib.linux-x86_64-2.7/anarci/Bio/Align

copying lib/python/anarci/Bio/Align/Generic.py -> build/lib.linux-x86_64-2.7/anarci/Bio/Align

copying lib/python/anarci/Bio/Align/__init__.py -> build/lib.linux-x86_64-2.7/anarci/Bio/Align

creating build/lib.linux-x86_64-2.7/anarci/Bio/Alphabet

copying lib/python/anarci/Bio/Alphabet/IUPAC.py -> build/lib.linux-x86_64-2.7/anarci/Bio/Alphabet

copying lib/python/anarci/Bio/Alphabet/__init__.py -> build/lib.linux-x86_64-2.7/anarci/Bio/Alphabet

copying lib/python/anarci/Bio/Alphabet/Reduced.py -> build/lib.linux-x86_64-2.7/anarci/Bio/Alphabet

creating build/lib.linux-x86_64-2.7/anarci/Bio/Data

copying lib/python/anarci/Bio/Data/SCOPData.py -> build/lib.linux-x86_64-2.7/anarci/Bio/Data

copying lib/python/anarci/Bio/Data/__init__.py -> build/lib.linux-x86_64-2.7/anarci/Bio/Data

copying lib/python/anarci/Bio/Data/IUPACData.py -> build/lib.linux-x86_64-2.7/anarci/Bio/Data

copying lib/python/anarci/Bio/Data/CodonTable.py -> build/lib.linux-x86_64-2.7/anarci/Bio/Data

creating build/lib.linux-x86_64-2.7/anarci/Bio/_py3k

copying lib/python/anarci/Bio/_py3k/__init__.py -> build/lib.linux-x86_64-2.7/anarci/Bio/_py3k

copying lib/python/anarci/Bio/_py3k/_ordereddict.py -> build/lib.linux-x86_64-2.7/anarci/Bio/_py3k

creating build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO

copying lib/python/anarci/Bio/SearchIO/FastaIO.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO

copying lib/python/anarci/Bio/SearchIO/__init__.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO

copying lib/python/anarci/Bio/SearchIO/_index.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO

copying lib/python/anarci/Bio/SearchIO/BlatIO.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO

copying lib/python/anarci/Bio/SearchIO/_utils.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO

creating build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO/_model

copying lib/python/anarci/Bio/SearchIO/_model/hit.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO/_model

copying lib/python/anarci/Bio/SearchIO/_model/__init__.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO/_model

copying lib/python/anarci/Bio/SearchIO/_model/_base.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO/_model

copying lib/python/anarci/Bio/SearchIO/_model/hsp.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO/_model

copying lib/python/anarci/Bio/SearchIO/_model/query.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO/_model

creating build/lib.linux-x86_64-2.7/anarci/Bio/SeqIO

copying lib/python/anarci/Bio/SeqIO/FastaIO.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SeqIO

copying lib/python/anarci/Bio/SeqIO/_convert.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SeqIO

copying lib/python/anarci/Bio/SeqIO/__init__.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SeqIO

copying lib/python/anarci/Bio/SeqIO/Interfaces.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SeqIO

copying lib/python/anarci/Bio/SeqIO/_index.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SeqIO

creating build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO/HmmerIO

copying lib/python/anarci/Bio/SearchIO/HmmerIO/hmmer3_tab.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO/HmmerIO

copying lib/python/anarci/Bio/SearchIO/HmmerIO/hmmer2_text.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO/HmmerIO

copying lib/python/anarci/Bio/SearchIO/HmmerIO/hmmer3_text.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO/HmmerIO

copying lib/python/anarci/Bio/SearchIO/HmmerIO/__init__.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO/HmmerIO

copying lib/python/anarci/Bio/SearchIO/HmmerIO/hmmer3_domtab.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO/HmmerIO

copying lib/python/anarci/Bio/SearchIO/HmmerIO/_base.py -> build/lib.linux-x86_64-2.7/anarci/Bio/SearchIO/HmmerIO

creating build/lib.linux-x86_64-2.7/anarci/Bio/AlignIO

copying lib/python/anarci/Bio/AlignIO/FastaIO.py -> build/lib.linux-x86_64-2.7/anarci/Bio/AlignIO

copying lib/python/anarci/Bio/AlignIO/ClustalIO.py -> build/lib.linux-x86_64-2.7/anarci/Bio/AlignIO

copying lib/python/anarci/Bio/AlignIO/NexusIO.py -> build/lib.linux-x86_64-2.7/anarci/Bio/AlignIO

copying lib/python/anarci/Bio/AlignIO/EmbossIO.py -> build/lib.linux-x86_64-2.7/anarci/Bio/AlignIO

copying lib/python/anarci/Bio/AlignIO/__init__.py -> build/lib.linux-x86_64-2.7/anarci/Bio/AlignIO

copying lib/python/anarci/Bio/AlignIO/Interfaces.py -> build/lib.linux-x86_64-2.7/anarci/Bio/AlignIO

copying lib/python/anarci/Bio/AlignIO/StockholmIO.py -> build/lib.linux-x86_64-2.7/anarci/Bio/AlignIO

copying lib/python/anarci/Bio/AlignIO/PhylipIO.py -> build/lib.linux-x86_64-2.7/anarci/Bio/AlignIO

running build_scripts

creating build/scripts-2.7

copying and adjusting bin/ANARCI -> build/scripts-2.7

copying bin/muscle -> build/scripts-2.7

copying bin/muscle_macOS -> build/scripts-2.7

changing mode of build/scripts-2.7/ANARCI from 644 to 755

changing mode of build/scripts-2.7/muscle from 700 to 755

changing mode of build/scripts-2.7/muscle_macOS from 700 to 755

running install_lib

running install_scripts

copying build/scripts-2.7/ANARCI -> /usr/local/bin

changing mode of /usr/local/bin/ANARCI to 755

changing mode of /usr/local/bin/muscle_macOS to 755

changing mode of /usr/local/bin/muscle to 755

running install_egg_info

Removing /usr/local/lib/python2.7/dist-packages/anarci-1.3.egg-info

Writing /usr/local/lib/python2.7/dist-packages/anarci-1.3.egg-info

Downloading germlines from IMGT and building HMMs...

 

RUN_pipeline.sh: line 8: python: command not found

RUN_pipeline.sh: line 13: python: command not found

Alignment input open failed.

   couldn't open ./curated_alignments/ALL.stockholm for reading

 

Error: File existence/permissions problem in trying to open HMM file ./HMMs/ALL.hmm.

HMM file ./HMMs/ALL.hmm not found (nor an .h3m binary of it)

 

 

Traceback (most recent call last):

  File "setup.py", line 75, in <module>

    shutil.copy( "curated_alignments/germlines.py", ANARCI_LOC )

  File "/usr/lib/python2.7/shutil.py", line 139, in copy

    copyfile(src, dst)

  File "/usr/lib/python2.7/shutil.py", line 96, in copyfile

    with open(src, 'rb') as fsrc:

IOError: [Errno 2] No such file or directory: 'curated_alignments/germlines.py'

 

Please see and let me know if you have any suggestions to successfully install ANARCI locally.

 

Thank you
```

**Comment by alirizaaribas-ibg** [comment]
Anyone solved this issue? My problem is the same:

```
python setup.py install --user
running install
/home/kalelie/miniconda3/lib/python3.9/site-packages/setuptools/command/install.py:34: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
/home/kalelie/miniconda3/lib/python3.9/site-packages/setuptools/command/easy_install.py:144: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
  warnings.warn(
running bdist_egg
running egg_info
writing anarci.egg-info/PKG-INFO
writing dependency_links to anarci.egg-info/dependency_links.txt
writing top-level names to anarci.egg-info/top_level.txt
reading manifest file 'anarci.egg-info/SOURCES.txt'
adding license file 'LICENCE'
adding license file 'LICENCE_MUSCLE'
writing manifest file 'anarci.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
creating build
creating build/lib
creating build/lib/anarci
copying lib/python/anarci/__init__.py -> build/lib/anarci
copying lib/python/anarci/anarci.py -> build/lib/anarci
copying lib/python/anarci/germlines.py -> build/lib/anarci
copying lib/python/anarci/schemes.py -> build/lib/anarci
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/anarci
copying build/lib/anarci/__init__.py -> build/bdist.linux-x86_64/egg/anarci
copying build/lib/anarci/anarci.py -> build/bdist.linux-x86_64/egg/anarci
copying build/lib/anarci/germlines.py -> build/bdist.linux-x86_64/egg/anarci
copying build/lib/anarci/schemes.py -> build/bdist.linux-x86_64/egg/anarci
byte-compiling build/bdist.linux-x86_64/egg/anarci/__init__.py to __init__.cpython-39.pyc
byte-compiling build/bdist.linux-x86_64/egg/anarci/anarci.py to anarci.cpython-39.pyc
byte-compiling build/bdist.linux-x86_64/egg/anarci/germlines.py to germlines.cpython-39.pyc
byte-compiling build/bdist.linux-x86_64/egg/anarci/schemes.py to schemes.cpython-39.pyc
installing package data to build/bdist.linux-x86_64/egg
running install_data
creating build/bdist.linux-x86_64/egg/bin
copying bin/muscle -> build/bdist.linux-x86_64/egg/bin
copying bin/muscle_macOS -> build/bdist.linux-x86_64/egg/bin
creating build/bdist.linux-x86_64/egg/EGG-INFO
installing scripts to build/bdist.linux-x86_64/egg/EGG-INFO/scripts
running install_scripts
running build_scripts
creating build/scripts-3.9
copying and adjusting bin/ANARCI -> build/scripts-3.9
changing mode of build/scripts-3.9/ANARCI from 664 to 775
creating build/bdist.linux-x86_64/egg/EGG-INFO/scripts
copying build/scripts-3.9/ANARCI -> build/bdist.linux-x86_64/egg/EGG-INFO/scripts
changing mode of build/bdist.linux-x86_64/egg/EGG-INFO/scripts/ANARCI to 775
copying anarci.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying anarci.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying anarci.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying anarci.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
zip_safe flag not set; analyzing archive contents...
anarci.__pycache__.anarci.cpython-39: module references __file__
creating 'dist/anarci-1.3-py3.9.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing anarci-1.3-py3.9.egg
removing '/home/kalelie/.local/lib/python3.9/site-packages/anarci-1.3-py3.9.egg' (and everything under it)
creating /home/kalelie/.local/lib/python3.9/site-packages/anarci-1.3-py3.9.egg
Extracting anarci-1.3-py3.9.egg to /home/kalelie/.local/lib/python3.9/site-packages
anarci 1.3 is already the active version in easy-install.pth
Installing ANARCI script to /home/kalelie/.local/bin

Installed /home/kalelie/.local/lib/python3.9/site-packages/anarci-1.3-py3.9.egg
Processing dependencies for anarci==1.3
Finished processing dependencies for anarci==1.3
Downloading germlines from IMGT and building HMMs...

Formatting alignments

IGs

TRs

Traceback (most recent call last):
  File "/archive/kalelie/ANARCI/build_pipeline/./RipIMGT.py", line 183, in <module>
    main()
  File "/archive/kalelie/ANARCI/build_pipeline/./RipIMGT.py", line 178, in main
    if ripfasta(species, gene_type, force = False):
  File "/archive/kalelie/ANARCI/build_pipeline/./RipIMGT.py", line 157, in ripfasta
    htmlfile = get_html(species, gene_type, force)
  File "/archive/kalelie/ANARCI/build_pipeline/./RipIMGT.py", line 138, in get_html
    if urllib.request.urlretrieve( urls[gene_type]%species,  filename ):
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 239, in urlretrieve
    with contextlib.closing(urlopen(url, data)) as fp:
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 214, in urlopen
    return opener.open(url, data, timeout)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 555, in error
    result = self._call_chain(*args)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 747, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 555, in error
    result = self._call_chain(*args)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 747, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 555, in error
    result = self._call_chain(*args)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 747, in http_error_302
    return self.parent.open(new, timeout=req.timeout)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 523, in open
    response = meth(req, response)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 632, in http_response
    response = self.parent.error(
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 561, in error
    return self._call_chain(*args)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 494, in _call_chain
    result = func(*args)
  File "/home/kalelie/miniconda3/lib/python3.9/urllib/request.py", line 641, in http_error_default
    raise HTTPError(req.full_url, code, msg, hdrs, fp)
urllib.error.HTTPError: HTTP Error 502: Proxy Error
Traceback (most recent call last):
  File "/archive/kalelie/ANARCI/build_pipeline/./FormatAlignments.py", line 480, in <module>
    main()
  File "/archive/kalelie/ANARCI/build_pipeline/./FormatAlignments.py", line 454, in main
    jalignments = format_j_genes(jalignments)
  File "/archive/kalelie/ANARCI/build_pipeline/./FormatAlignments.py", line 183, in format_j_genes
    aligned = read_fasta( al_filename )
  File "/archive/kalelie/ANARCI/build_pipeline/./FormatAlignments.py", line 127, in read_fasta
    handle = open(filename, "r")
FileNotFoundError: [Errno 2] No such file or directory: '/archive/kalelie/ANARCI/build_pipeline/./muscle_alignments/all_js_aligned.fasta'
Alignment input open failed.
   couldn't open ./curated_alignments/ALL.stockholm for reading

Error: File existence/permissions problem in trying to open HMM file ./HMMs/ALL.hmm.
HMM file ./HMMs/ALL.hmm not found (nor an .h3m binary of it)


Traceback (most recent call last):
  File "/archive/kalelie/ANARCI/setup.py", line 61, in <module>
    shutil.copy( "curated_alignments/germlines.py", ANARCI_LOC )
  File "/home/kalelie/miniconda3/lib/python3.9/shutil.py", line 427, in copy
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/home/kalelie/miniconda3/lib/python3.9/shutil.py", line 264, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: 'curated_alignments/germlines.py'
```
Also it was installed successfully on hpc two weeks ago. Always python2 is working despite python 3 is installed in new environment on Centos 7.9. How to overcome python versions? I tried `alias` but nothing changed.

**Comment by dweb0** [comment]
I was having the same problem (CentOS Stream 8, python3.9). It turned out `bin/muscle`, was not getting added to my PATH during setup. As a workaround, I changed line 181 of `build_pipeline/FormatAlignments.py` to point to the included `muscle` executable from the repository, instead of using whatever `muscle` is in PATH.

```
pr = Popen( [ "/home/dweb0/ANARCI/bin/muscle", "-in", ffile, "-gapopen", "-10", "-out", al_filename, ], stdout=PIPE, stderr=PIPE )
```

Before that, I had tried installing the latest version of `muscle` (5.1) to my PATH, but it seems that version produces output not compatible with what is required in this build pipeline.
